internal/dkg/store.go

```diff
@@ -1,181 +1,117 @@
 
-const BoltFileName = "dkg.db"
-const BoltStoreOpenPerm = 0660
 const DirPerm = 0755
 
-var stagedStateBucket = []byte("dkg")
-var finishedStateBucket = []byte("dkg_finished")
+type FileStore struct {
+	baseFolder string
+	log        log.Logger
+}
 
-func NewDKGStore(baseFolder string, options *bolt.Options) (*BoltStore, error) {
+func (fs *FileStore) getDKGFolder(beaconID string) string {
+	return path.Join(fs.baseFolder, beaconID, StoreFolder)
+}
+
+func NewDKGStore(baseFolder string, logLevel int) (*FileStore, error) {
 	err := os.MkdirAll(baseFolder, DirPerm)
 	if err != nil {
 		return nil, err
 	}
-	dbPath := path.Join(baseFolder, BoltFileName)
-	db, err := bolt.Open(dbPath, BoltStoreOpenPerm, options)
-	if err != nil {
-		return nil, err
-	}
-
-	err = db.Update(func(tx *bolt.Tx) error {
-		_, err := tx.CreateBucketIfNotExists(stagedStateBucket)
-		if err != nil {
-			return err
-		}
+	return &FileStore{
+		baseFolder: baseFolder,
+		log:        log.New(nil, logLevel, true),
+	}, nil
+}
 
-		_, err = tx.CreateBucketIfNotExists(finishedStateBucket)
-		return err
-	})
+func getFromFilePath(path string) (*DBState, error) {
+	t := DBStateTOML{}
+	_, err := toml.DecodeFile(path, &t)
 	if err != nil {
 		return nil, err
 	}
-
-	store := BoltStore{
-		db:  db,
-		log: log.New(nil, log.DebugLevel, true),
-	}
-
-	return &store, nil
-}
-
-func (s *BoltStore) GetCurrent(beaconID string) (*DBState, error) {
-	dkg, err := s.get(beaconID, stagedStateBucket)
-
+	state, err := t.FromTOML()
 	if err != nil {
 		return nil, err
 	}
+	return state, nil
+}
 
-	if dkg == nil {
+func (fs *FileStore) GetCurrent(beaconID string) (*DBState, error) {
+	f, err := getFromFilePath(path.Join(fs.getDKGFolder(beaconID), StagedFileName))
+	if errors.Is(err, os.ErrNotExist) {
+		fs.log.Debug("No DKG file found, returning new state")
 		return NewFreshState(beaconID), nil
 	}
-	return dkg, nil
+	return f, err
 }
 
-func (s *BoltStore) GetFinished(beaconID string) (*DBState, error) {
-	return s.get(beaconID, finishedStateBucket)
-}
-
-func (s *BoltStore) get(beaconID string, bucketName []byte) (*DBState, error) {
-	var dkg *DBState
-
-	err := s.db.View(func(tx *bolt.Tx) error {
-		bucket := tx.Bucket(bucketName)
-		if bucket == nil {
-			return errors.Errorf("%s bucket was nil - this should never happen", bucketName)
-		}
-		value := bucket.Get([]byte(beaconID))
-		if value == nil {
-			return nil
-		}
-		t := DBStateTOML{}
-		_, err := toml.NewDecoder(bytes2.NewReader(value)).Decode(&t)
-		if err != nil {
-			return err
-		}
-
-		d, err := t.FromTOML()
-		if err != nil {
-			return err
-		}
-		dkg = d
-		return nil
-	})
-
-	return dkg, err
-}
-
-func (s *BoltStore) SaveCurrent(beaconID string, state *DBState) error {
-	return s.save(stagedStateBucket, beaconID, state)
+func (fs *FileStore) GetFinished(beaconID string) (*DBState, error) {
+	f, err := getFromFilePath(path.Join(fs.getDKGFolder(beaconID), FileName))
+	if errors.Is(err, os.ErrNotExist) {
+		return nil, nil
```

Comment:
in fairness this is how it was before. In principle, having no `Finished` DKG isn't an error. I think returning `os.ErrNotExist` from here would be a leaky abstraction (because it's behind an interface that may or may not be a file).

I'd be open to remapping to a custom err such as `ErrNoFinishedDKG` here, but I don't think it's that meaningful over `nil, nil`

Reply:
Yeah what Patrick said, this is a drop in replacement matching the behavior of the last implementation